File: guns-admin/src/main/webapp/static/modular/product/product/product.js
/**
 * 产品管理初始化
 */
var Product = {
    id: "ProductTable",	//表格id
    seItem: null,		//选中的条目
    table: null,
    layerIndex: -1
};

/**
 * 初始化表格的列
 */
Product.initColumn = function () {
    return [
        {field: 'selectItem', radio: true},
//            {title: '租户id', field: 'tenantId', visible: true, align: 'center', valign: 'middle'},
//            {title: '产品ID', field: 'productId', visible: true, align: 'center', valign: 'middle'},
            {title: '产品PK', field: 'productKey', visible: true, align: 'center', valign: 'middle'},
//            {title: '数据格式', field: 'dataFormat', visible: true, align: 'center', valign: 'middle'},
//            {title: '入网类型', field: 'netType', visible: true, align: 'center', valign: 'middle'},
//            {title: '产品密钥', field: 'productSecret', visible: true, align: 'center', valign: 'middle'},
//            {title: '节点类型', field: 'nodeType', visible: true, align: 'center', valign: 'middle'},
//            {title: '领域', field: 'domain', visible: true, align: 'center', valign: 'middle'},
            {title: '产品名称', field: 'name', visible: true, align: 'center', valign: 'middle'},
//            {title: '地域', field: 'region', visible: true, align: 'center', valign: 'middle'},
//            {title: '所有者领域', field: 'ownerDomain', visible: true, align: 'center', valign: 'middle'},
//            {title: '归属品类id', field: 'categoryld', visible: true, align: 'center', valign: 'middle'},
//            {title: 'categoryKey', field: 'categoryKey', visible: true, align: 'center', valign: 'middle'},
//            {title: 'categoryName', field: 'categoryName', visible: true, align: 'center', valign: 'middle'},
//            {title: '访问方式', field: 'accessMethod', visible: true, align: 'center', valign: 'middle'},
            {title: 'IoT平台', field: 'iotPackage', visible: true, align: 'center', valign: 'middle'},
            {title: '产品状态', field: 'status', visible: true, align: 'center', valign: 'middle'},
            {title: '创建者', field: 'creator', visible: true, align: 'center', valign: 'middle'},
            {title: '创建时间', field: 'gmtCreate', visible: true, align: 'center', valign: 'middle'},
            {title: '修改者', field: 'modifier', visible: true, align: 'center', valign: 'middle'},
            {title: '修改时间', field: 'gmtModified', visible: true, align: 'center', valign: 'middle'}
//            {title: '商品码', field: 'aliyunCommodityCode', visible: true, align: 'center', valign: 'middle'},
//            {title: 'connectMode', field: 'connectMode', visible: true, align: 'center', valign: 'middle'},
//            {title: 'rbac租户Id', field: 'rbacTenantId', visible: true, align: 'center', valign: 'middle'},
    ];
};

/**
 * 检查是否选中
 */
Product.check = function () {
    var selected = $('#' + this.id).bootstrapTable('getSelections');
    if(selected.length == 0){
        Feng.info("请先选中表格中的某一记录！");
        return false;
    }else{
        Product.seItem = selected[0];
        return true;
    }
};

/**
 * 点击添加产品
 */
Product.openAddProduct = function () {
    var index = layer.open({
        type: 2,
        title: '添加产品',
        area: ['800px', '420px'], //宽高
        fix: false, //不固定
        maxmin: true,
        content: Feng.ctxPath + '/product/product_add'
    });
    this.layerIndex = index;
};
/**
 * 更新产品详情
 */
Product.updateProductDetail = function () {
	
}

/**
 * 打开查看产品详情
 */
Product.openProductDetail = function () {
    if (this.check()) {
        var index = layer.open({
            type: 2,
            title: '产品详情',
            area: ['800px', '420px'], //宽高
            fix: false, //不固定
            maxmin: true,
            content: Feng.ctxPath + '/product/product_update/' + Product.seItem.id
        });
        this.layerIndex = index;
    }
};

/**
 * 删除产品
 */
Product.delete = function () {
    if (this.check()) {
        var ajax = new $ax(Feng.ctxPath + "/product/delete", function (data) {
            Feng.success("删除成功!");
            Product.table.refresh();
        }, function (data) {
            Feng.error("删除失败!" + data.responseJSON.message + "!");
        });
        ajax.set("productId",this.seItem.id);
        ajax.start();
    }
};

/**
 * 查询产品列表
 */
Product.search = function () {
    var queryData = {};
    queryData['condition'] = $("#condition").val();
    Product.table.refresh({query: queryData});
};

$(function () {
    var defaultColunms = Product.initColumn();
    var table = new BSTable(Product.id, "/product/list", defaultColunms);
    table.setPaginationType("client");
    Product.table = table.init();
});

File: guns-admin/src/main/java/com/stylefeng/guns/modular/custom/model/Product.java
package com.stylefeng.guns.modular.custom.model;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

import com.baomidou.mybatisplus.activerecord.Model;
import com.baomidou.mybatisplus.annotations.TableField;
import com.baomidou.mybatisplus.annotations.TableId;
import com.baomidou.mybatisplus.annotations.TableName;
import com.baomidou.mybatisplus.enums.IdType;
import lombok.Data;

/**
 * <p>
 * 产品表
 * </p>
 *
 * @author guanqing123
 * @since 2018-07-12
 */
@Data
@TableName("app_product")
public class Product extends Model<Product> {

    private static final long serialVersionUID = 1L;

    /**
     * 租户id
     */
    private Long tenantId;
    /**
     * 产品PK
     */
	@TableId(value="productKey", type= IdType.INPUT)
    private String productKey;
    /**
     * 数据格式
     */
    private String dataFormat;
    /**
     * 入网类型
     */
    private String netType;
    /**
     * 产品密钥
     */
    private String productSecret;
    /**
     * 节点类型
     */
    private String nodeType;
    /**
     * 领域
     */
    private String domain;
    /**
     * 产品名称
     */
    private String name;
    /**
     * 地域
     */
    private String region;
    /**
     * 所有者领域
     */
    private String ownerDomain;
    /**
     * 归属品类id
     */
    private Long categoryld;
    /**
     * 归属品类Key
     */
    private String categoryKey;
    /**
     * 归属品类Name
     */
    private String categoryName;
    /**
     * 访问方式
     */
    private String accessMethod;
    /**
     * 产品状态
     */
    private String status;
    /**
     * 创建时间
     */
    private Date gmtCreate;
    /**
     * 修改时间
     */
    private Date gmtModified;
    /**
     * 创建者
     */
    private String creator;
    /**
     * 修改者
     */
    private String modifier;
    /**
     * 产品ID
     */
    private Long productId;
    /**
     * 商品码
     */
    private String aliyunCommodityCode;
    /**
     * 连接模式
     */
    private String connectMode;
    /**
     * RBAC租户Id
     */
    private String rbacTenantId;
    /**
     * IoT平台
     */
    private Integer iotPackage;
    /**
     * useId2Auth
     */
    private Boolean useId2Auth;
    /**
     * 设备数量
     */
    private Integer deviceNumLimit;
    
    @TableField(exist=false)
    private List<ProductExtend> extendProperties;

	@Override
    protected Serializable pkVal() {
        return this.productKey;
    }

    @Override
    public String toString() {
        return "Product{" +
        "tenantId=" + tenantId +
        ", productKey=" + productKey +
        ", dataFormat=" + dataFormat +
        ", netType=" + netType +
        ", productSecret=" + productSecret +
        ", nodeType=" + nodeType +
        ", domain=" + domain +
        ", name=" + name +
        ", region=" + region +
        ", ownerDomain=" + ownerDomain +
        ", categoryld=" + categoryld +
        ", categoryKey=" + categoryKey +
        ", categoryName=" + categoryName +
        ", accessMethod=" + accessMethod +
        ", status=" + status +
        ", gmtCreate=" + gmtCreate +
        ", gmtModified=" + gmtModified +
        ", creator=" + creator +
        ", modifier=" + modifier +
        ", productId=" + productId +
        ", aliyunCommodityCode=" + aliyunCommodityCode +
        ", connectMode=" + connectMode +
        ", rbacTenantId=" + rbacTenantId +
        ", iotPackage=" + iotPackage +
        ", useId2Auth=" + useId2Auth +
        ", deviceNumLimit=" + deviceNumLimit +
        "}";
    }
}
